### ISSUE TITLE
Multiple `gfx` crates make the example not build.

### ISSUE THREAD

**Issue by csherratt** [gfx-rs/gfx-rs#610]
```
Compiling gfx v0.2.0 (file:///home/travis/build/gfx-rs/gfx-rs)
Compiling gfx v0.2.0 (https://github.com/gfx-rs/gfx-rs#79e00b2e)
```

The problem is the examples are pulling in `gfx_device_gl` which pulls its own copy of `gfx` from github.

Do we move the examples out of `gfx`? like `PistionExamples`; or is there another way to make this work?

**Comment by kvark** [comment]
cc @bvssvni 

I suppose we should move them out.

**Comment by kvark** [comment]
What we could have is a device-independent repository of examples, which are provided as a library. Then, each device (let's say `gfx_device_gl` and `gfx_device_soft`) will have its own examples which create a window and a device, and then just call the common examples to use it.

**Comment by kvark** [comment]
Fixed by #598
